src/otx/core/data/dataset/base.py

```diff
@@ -92,8 +94,9 @@ def _sample_another_idx(self) -> int:
         return np.random.default_rng().integers(0, len(self))
 
     def _apply_transforms(self, entity: T_OTXDataEntity) -> T_OTXDataEntity | None:
-        if isinstance(self.transforms, Compose):
+        if self.to_tv_image:
             entity = entity.to_tv_image()
+        if isinstance(self.transforms, Compose):
             return self.transforms(entity)
```

Comment:
```suggestion
        if isinstance(self.transforms, Compose):
            if self.to_tv_image:
                entity = entity.to_tv_image()
            return self.transforms(entity)
```
Since 'entity = entity.to_tv_image()' was only applied to mmdet pipeline and not to other pipelines, it seems risky to place it outside of the Compose-related if statement.